test/integration/scheduler/preemption_test.go

```diff
@@ -129,6 +131,51 @@ var _ = ginkgo.Describe("Preemption", func() {
 			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, cq.Name, highWl2)
 			util.ExpectWorkloadsToBePending(ctx, k8sClient, lowWl1, lowWl2)
 		})
+
+		ginkgo.It("Should preempt newer Workloads with the same priority when there is not enough quota", func() {
+			ginkgo.By("Creating initial Workloads")
+			wl1 := testing.MakeWorkload("wl-1", ns.Name).
+				Queue(q.Name).
+				Priority(lowPriority).
+				Request(corev1.ResourceCPU, "1").
+				Obj()
+			wl2 := testing.MakeWorkload("wl-2", ns.Name).
+				Queue(q.Name).
+				Priority(lowPriority).
+				Request(corev1.ResourceCPU, "1").
+				Obj()
+			wl3 := testing.MakeWorkload("wl-3", ns.Name).
+				Queue(q.Name).
+				Priority(lowPriority).
+				Request(corev1.ResourceCPU, "3").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl1)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, wl2)).To(gomega.Succeed())
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, cq.Name, wl1, wl2)
+
+			gomega.Expect(k8sClient.Create(ctx, wl3)).To(gomega.Succeed())
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, wl3)
+
+			ginkgo.By("Waiting one second, to ensure that the new workload has a later creation time")
+			time.Sleep(time.Second)
+
+			ginkgo.By("Creating a new Workload")
+			wl4 := testing.MakeWorkload("wl-4", ns.Name).
+				Queue(q.Name).
+				Priority(lowPriority).
+				Request(corev1.ResourceCPU, "1").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl4)).To(gomega.Succeed())
```

Comment:
Uhm... Updating the `CreationTimestamp` in the wl4 might be better than creating the wl4 after waiting for a second. Since the key here is not to wait for other components (e.g., scheduler) but whether the `CreationTimestamp` is newer.

@trasc @alculquicondor wdyt?

Reply:
Granted that waiting is not ideal, but I don't know if the control plane started by envtest will allow it, and even if it is allowed it's something that we don't expect to happen with a real api server.

Reply:
+1 on @trasc
At least 1s is not that much for an integration test.

Reply:
>  we don't expect to happen with a real api server.

You're right. thanks.